File: 922-sort-array-by-parity-ii/922-sort-array-by-parity-ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParityII = function(nums) {
    let oddArr = [];
    let evenArr = [];
    
    for(let i = 0; i < nums.length; i++){
        if(nums[i] % 2 === 0){
            evenArr.push(nums[i]);        
        }else{
            oddArr.push(nums[i]);
        }
    }
    
    let ansArr = [];
    
    for(let i = 0; i < evenArr.length; i++){
        ansArr.push(evenArr[i]);
        ansArr.push(oddArr[i]);
    }
    
    return ansArr;
};

File: 922-sort-array-by-parity-ii/922-sort-array-by-parity-ii.go
func sortArrayByParityII(nums []int) []int {
    evenArr := []int{}
    oddArr := []int{}
    
    for i := 0; i < len(nums); i++{
        if nums[i] % 2 == 0{
            evenArr = append(evenArr, nums[i])
        }else{
            oddArr = append(oddArr, nums[i])
        }
    }
    
    ansArr := []int{}
    
    for i := 0; i < len(evenArr); i++{
        ansArr = append(ansArr, evenArr[i])
        ansArr = append(ansArr, oddArr[i])
    }
    
    return ansArr
}